### PR TITLE
Build Environment Fixes

### PR DESCRIPTION
**Pull request by oetiker**
## What do these changes do?

* add swarm support for OSX
* workaround incompatibilities between `docker-compose` and `docker stack deploy` by having the config files parse by composing first and then feeding the parsed and combined files to docker ... this aproach resolves the following annoying incompatibilities:
  * loading of .env files
  * inconsistent handling of command overrides
* this also removes the hardcoded dns entries since this prevents the swarm from working anywhere outside itis. Make sure your host has proper /etc/resolve.conf OR set the dns key in the daemon.json file of your local docker instead.

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] **Runs in the swarm**
- [ ] If you design a new module, add your user to .github/CODEOWNERS

[waffle.io]:https://waffle.io/marketing-assets/documents/waffleio_cheatsheet_v1.pdf?utm_source=blog&utm_medium=cheatsheet-ctabutton&utm_campaign=cheatsheet

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,6 +18,13 @@ export DOCKER=docker.exe
 export RUN_DOCKER_ENGINE_ROOT=1
 export DOCKER_GID=1042
 export HOST_GID=1000
+else ifneq (,$(findstring Darwin,$(VERSION)))
+$(info    detected OSX)
+export DOCKER_COMPOSE=docker-compose
+export DOCKER=docker
+export RUN_DOCKER_ENGINE_ROOT=1
+export DOCKER_GID=1042
```
Comment (pcrespov) [inline Makefile]: these variables are already defined in .venv-devel, oder?

Comment (sanderegg) [reply Makefile]: DOCKER_GID is defined, but I guess a linux user might want to override them with whatever is defined on his/her machine?
RUN_DOCKER_ENGINE_ROOT is defined as 0, but needs to be 1 for any non-linux host.

Comment (oetiker) [reply Makefile]: this PR just adds support for osx ... lets fix the deeper inconsistencies in a separate one

**Comment by oetiker** [comment]
new approach at resolving the swarm issues

File: Makefile [hunk 2]
```diff
@@ -60,14 +75,23 @@ up:
 
 up-swarm:
 	${DOCKER} swarm init
-	${DOCKER} stack deploy -c services/docker-compose.yml -c services/docker-compose.deploy.yml  -c services/docker-compose.tools.yml services
+	${DOCKER_COMPOSE} -f services/docker-compose.yml -f services/docker-compose.deploy.yml -f services/docker-compose.tools.yml config > $(TEMPCOMPOSE).tmp-compose.yml ; ${DOCKER} stack deploy -c $(TEMPCOMPOSE).tmp-compose.yml services; rm $(TEMPCOMPOSE).tmp-compose.yml
```
Comment (sanderegg) [inline Makefile]: This looks pretty! but why not on several lines? I almost thought you replaced the stack by a compose...

**Pull request merged**